src/System.Diagnostics.Tracing.Telemetry/src/System/Diagnostics/Tracing/TelemetrySource.cs

```diff
@@ -0,0 +1,41 @@
+using System;
+
+namespace System.Diagnostics.Tracing
+{
+    /// <summary>
+    /// This is the basic API to 'hook' parts of the framework.   It is like an EventSource
+    /// (which can also write object), but is intended to log complex objects that can't be serialized.
+    /// </summary>
+    public abstract class TelemetrySource
```

Comment:
Rather than having the data be of type ```object```, was any consideration given to have these types be generic and having the data be of type ```T```, to avoid boxing costs for example when tracing out value types?  I realize that could complicate the type hierarchy and a goal for having a base non-generic for these, but maybe the complexity would be worth it?  Or do we not expect ```parameters``` to be value types or for these to be called with any impactful frequency?